src/build123d/__init__.py

```diff
@@ -163,6 +163,12 @@
     "delta",
     "new_edges",
     "edges_to_wires",
+    # Context aware selectors
+    "solids",
+    "faces",
+    "wires",
+    "edges",
+    "vertices",
```

Comment:
To make this complete, there should be `solid`, `face`, `wire`, `edge` and `vertex` versions as well.

Reply:
Would this be ok then? Much less to maintain since its just wrapping the existing functions.
```py
def __gen_context_component_getter(f: Callable):
    @functools.wraps(f)
    def getter(select: Select = Select.ALL):
        context = Builder._get_context(f.__name__)
        if not context: raise RuntimeError(f"{f.__name__}() requires a Builder context to be in scope")
        return f(context, select)
    return getter

vertices = __gen_context_component_getter(Builder.vertices)
edges = __gen_context_component_getter(Builder.edges)
wires = __gen_context_component_getter(Builder.wires)
faces = __gen_context_component_getter(Builder.faces)
solids = __gen_context_component_getter(Builder.solids)

vertex = __gen_context_component_getter(Builder.vertex)
edge = __gen_context_component_getter(Builder.edge)
wire = __gen_context_component_getter(Builder.wire)
face = __gen_context_component_getter(Builder.face)
solid = __gen_context_component_getter(Builder.solid)
```

Reply:
It's a pleasure working with skilled programmers like yourself, I would never have come up with this solution but it looks great.  I was worried that the sphinx documentation system would have trouble with this but it's happy. Go ahead with the PR and don't worry about the docs, I'll add these new functions.

Reply:
Likewise!

Reply:
Likewise!